lc141_linked_list_cycle.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+
+using namespace std;
+
+ struct ListNode {
+     int val;
+     ListNode *next;
+     ListNode(int x) : val(x), next(NULL) {}
+ };
+class Solution {
+	public:
+		bool hasCycle(ListNode *head) {
+			if (!head)
+				return false;
+			if(head == head->next)
+				return true;
+			if(!head->next || !head->next->next)
+				return false;
+			ListNode *pos = head, *double_pos = head->next->next;
+
+			while(double_pos){
+				if(pos == double_pos)
+					return true;
+				pos = pos->next;
+				if(!double_pos->next || !double_pos->next->next)
+					return false;
+				double_pos = double_pos->next->next;
+			}
+			return false;
```

Comment:
Not sure if you'll hit this line ever!

Reply:
I'm not sure, which line are you talking about. If it is line number 29, then I think you are right. I can drop the final 'return false' statement.